pkg/registry/interfaces.go

```diff
@@ -14,16 +14,19 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package genericapiserver
+package registry
 
 import (
+	"k8s.io/kubernetes/pkg/genericapiserver"
 	"k8s.io/kubernetes/pkg/registry/generic"
 	"k8s.io/kubernetes/pkg/runtime/schema"
 )
 
+// RESTOptionsGetter is a function type for returning RESTOptions for a GroupVersion.
 type RESTOptionsGetter func(resource schema.GroupResource) generic.RESTOptions
 
+// RESTStorageProvider is a factory type for REST storage.
 type RESTStorageProvider interface {
```

Comment:
All the other uses are just type assertions, right?  You can delete them and our usage will keep them in line.